src/DynamoCoreWpf/Utilities/LibraryDragAndDrop.cs

```diff
@@ -29,18 +29,24 @@ internal void HandleMouseDown(Point position, NodeSearchElementViewModel node)
 
         internal void HandleMouseMove(FrameworkElement sender, Point currentPosition)
         {
-            if (isDragging || nodeViewModel == null)
+            if (isDragging)
                 return;
 
             // If item was dragged enough, then fire DoDragDrop. 
             // Otherwise it means user click on item and there is no need to fire DoDragDrop.
             var deltaX = System.Math.Abs(currentPosition.X - startPosition.X);
             if (deltaX < SystemParameters.MinimumHorizontalDragDistance)
+            {
+                isDragging = false;
                 return;
+            }
 
             var deltaY = System.Math.Abs(currentPosition.Y - startPosition.Y);
             if (deltaY < SystemParameters.MinimumVerticalDragDistance)
+            {
+                isDragging = false;
                 return;
+            }
```

Comment:
Sorry, but I did not get the point here. If the mouse move is too small, then the drag operation is cancelled right away - I think that is the original behavior. So when a node is dragged to canvas, and on mouse release the node will be added to the canvas. If not the drag operation is cancelled right away. 
Before this fix, the tree assumes that there is still a drag operation (nodeviewmodel is not null) for every node element click. So, I added a check there on mouse move and made dragging to false, if the drag is with in that minimum distance. I can very well add nodeviewmodel == null, and remove the check on mousemove, but that brings another question - when will nodeviewmodel be null? For every drag we set nodeviewmodel, but once the drag is completed, we never set back to null. In that case, shouldn't we set nodeviewmodel to null once the drag is completed?  Again, another case here to consider here is, if the item is not dragged enough, that is item is not within that distance, today we dont fire drag and drop - but we shouldnt we set nodeviewmodel to null here? 

Or a simple fix will be remove the null check on OnExpanderButtonMouseLeftButtonDown. So nodeviewmodel will be set to null if there is no drag operation. 

@aosyatnik - any thoughts here?  Since you implemented drag and drop, you might have different solution. 
we can discuss this on friday your time.